### PR TITLE
gBoards ButterStick

### PR DESCRIPTION
**Pull request by germ**
Part two of the gboards-consolidation project. The Mega PR has been separated out into smaller per-keyboard PRs.

This PR contains a single keyboard as per @zvecr , in this case ButterStick. This code was pulled from #8262 

- [x] Enhancement/optimization
- [x] Keymap/layout/userspace (addition or update)

## Issues Fixed or Closed by This PR

- Replaces #8262
- Waiting on #8921

### CODE REVIEW

File: keyboards/gboards/k/butterstick/config.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x1337
```
Comment (drashna) [inline keyboards/gboards/k/butterstick/config.h:6]: This Vendor ID is assigned to Radiant Networks Plc.

https://www.the-sz.com/products/usbid/index.php?v=0x1337

Please change it.

File: keyboards/gboards/k/butterstick/rules.mk [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# QMK Options
+MOUSEKEY_ENABLE		= yes				# Mouse keys(+4700)
+EXTRAKEY_ENABLE		= yes       # Audio control and System control(+450)
+CONSOLE_ENABLE		= yes       # Console for debug(+400)
+COMMAND_ENABLE		= no				# Commands for debug and configuration
+NKRO_ENABLE				= yes       # USB Nkey Rollover
+
+# Extra defs
+OPT_DEFS 						+= -DDEBUG_MATRIX
+EXTRAFLAGS 					+= -flto
```
Comment (drashna) [inline keyboards/gboards/k/butterstick/rules.mk:14]:
```suggestion
LTO_ENABLE = yes
```

Comment (germ) [reply keyboards/gboards/k/butterstick/rules.mk]: Manually changed in https://github.com/qmk/qmk_firmware/pull/8922/commits/bb85f4cbe79eee1f8869105959890ee3f92906e7

File: keyboards/gboards/k/butterstick/config.h (hunk 1 above)
Comment (germ) [reply keyboards/gboards/k/butterstick/config.h]: Changed in https://github.com/qmk/qmk_firmware/pull/8922/commits/bb85f4cbe79eee1f8869105959890ee3f92906e7

**Comment by Erovia** [comment]
Could you move the files with git instead of manually?  
`mkdir keyboards/gboards; git mv keyboards/butterstick/ keyboards/gboards/` should do the trick.  
This way git know that the files were moved and not removed and new ones created.  
Also, could you please target the `develop` branch with this as it's a breaking one?

**Comment by germ** [comment]
I don't have time at the moment, nor an idea of what you mean by 'target the develop branch'. If you can rip that together into a different PR I'll sign off on it.

**Comment by drashna** [comment]
>Target the `develop` branch

Mostly, that means edit the title, and that should also allow you to change the target of the PR from qmk:master to a different branch.

**Comment by tzarc** [comment]
Can you please go through the following procedure -- I'm assuming you're using the normal setup with the real QMK repo set up as the `upstream`:
```sh
git fetch --all --tags
git checkout gboards-butterstick
git branch gboards-butterstick-old    # This makes sure you have an old copy of your branch
git reset --hard upstream/develop
git mv keyboards/butterstick keyboards/gboards/butterstick
git checkout gboards-butterstick-old -- keyboards/gboards/butterstick
git add keyboards/gboards/butterstick
```

This should get this branch up to speed, making it mergeable.
Unsure if `sten.c` and `sten.h` need removing -- will leave it up to you.

Once you're happy, commit any of your changes as normal, then:
```
git push origin gboards-butterstick --force-with-lease
```

You can use a similar procedure for the other boards of yours that have been split out.
Feel free to ping me if you've got issues, as well as once it's done so we can fast-track a merge.

**Comment by germ** [comment]
I should have time to split everything up from germ/qmk_firmware:master and get that pushed in a few separate PRs here. By the looks of it should have the common code for everything and then a relocation of my boards to the created vendor folder. That would need to target breaking changes, but should be acceptable (people have been using a crusty version of QMK for too long!)

If someone can take a quick prep through germ/qmk_firmware/k/* for any changes that would be prudent to apply before opening those PRs, it would save some trouble! Thanks!

**Comment by zvecr** [comment]
Thank you for your contribution!

This pull request has been automatically closed because it has not had activity in the last 30 days.
Please feel free to give a status update now, ping for review, or re-open when it's ready.

**Pull request closed without merging**